src/rpm/builder.rs

```diff
@@ -475,7 +475,7 @@ impl RPMBuilder {
     #[cfg(feature = "signature-meta")]
     pub fn build_and_sign<S>(self, signer: S) -> Result<RPMPackage, RPMError>
     where
-        S: signature::Signing<signature::algorithm::RSA>,
```

Comment:
I don't quite understand why we revert a compile time check back to a runtime check. Instead of removing it the `Algorithm` could have become a generic too.

Reply:
There was no compile-time check because you don't get to avoid the part where the key itself is of the type you want.  With `build_and_sign::<algorithm::RSA>` you still have to make sure the key you pass in is RSA, except that we would then need to return a runtime error when you pass in a key of the wrong type.

That just doesn't feel helpful at all.

Reply:
What I tried to say, if we can avoid runtime `match`es on the key type, and expect the user to provide the correct one, it allows for a more meta code representation approach, which I'd personally prefer over `match` in quite a few places. The compiler can then enforce a few more things and _users_ that know their key type for sure, can have the `match` _outside_ of `rpm-rs` and we can be happy OR possibly take advantage of the type based definition of key types _iff_ they can based on meta programming on their end, too. 

Reasons pro generic based crypto:

* user can use generics to extend type saftey
* compiler help
* less `match` based clutter
* avoids the `build_and_sign_...` variants which all almost do the same
* could separate concerns of doing the actual signing vs the mechanics of creating the tag

Reply:
Imho we should not _parse_ AND _sign_ in the same function, but provide key types that are external to the i.e. `build_and_sign` call / `Signer`.